### DISCUSSION TITLE
DuckDb JDBC InvalidInputException

### DISCUSSION THREAD

**Discussion by ruchirj** [duckdb/duckdb#6032] [General]
I am running into this error occasionally while using DuckDb with JDBC. This error results into the JVM crashing. The error sounds like there was a problem in executing the query but why does that result in the JVM crashing altogether? Also how do I get more info on what is the error about?

```
receiveTimestamp: "2023-01-30T00:27:24.893432825Z"
resource: {2}
severity: "ERROR"
textPayload: "  what():  Invalid Input Error: Attempting to execute an unsuccessful or closed pending query result"
timestamp: "2023-01-30T00:27:19.469106889Z"
}
```

**Comment by Alex-Monahan** [comment]
Hello! Could you please file an issue and follow the bug report template?  More information would be helpful so that the error can be reproduced!

**Comment by ruchirj** [comment]
@Alex-Monahan so I think I have it narrowed down to when I reuse a connection across multiple threads. The reason I am reusing connections is because otherwise as noted [here](https://github.com/duckdb/duckdb/discussions/5985#discussioncomment-4776448) the latency of resetting S3 credentials is high and @Mytherin suggested that I reuse connections and that helped with the latency bit but brought up this issue. Once I start creating 1 connection per query, the problem goes away.

**Reply by Mause** [reply]
You should be aiming to use one connection per thread, generally

**Reply by ruchirj** [reply]
Thanks @Mause, using one connection / thread does seem to have helped! Just curious why this is, since this is not in the JDBC spec. Also, this isn't great since I can't use fork-join executors or cached executors where thread reuse may be low and then we are back to the latency problem I referred to above.

**Reply by Mause** [reply]
Can you clarify what the latency you're referring to is?

**Reply by ruchirj** [reply]
[Here](https://github.com/duckdb/duckdb/discussions/5985#discussioncomment-4776448)
